src/main/java/AuthServlet.java

```diff
@@ -0,0 +1,77 @@
+import server.chat.daoimpl.UserFileDaoImpl;
+import server.chat.model.User;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+@WebServlet(name = "AuthServlet")
+public class AuthServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String uname = request.getParameter("username");
+        String password = request.getParameter("password");
+        String generatedPassword = null;
+        try {
+            // Create MessageDigest instance for MD5
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            //Add password bytes to digest
+            md.update(password.getBytes());
+            //Get the hash's bytes
+            byte[] bytes = md.digest();
+            //This bytes[] has bytes in decimal format;
+            //Convert it to hexadecimal format
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < bytes.length; i++) {
+                sb.append(Integer.toString((bytes[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            //Get complete hashed password in hex format
+            generatedPassword = sb.toString();
+        }
+        catch (NoSuchAlgorithmException e)
+        {
+            e.printStackTrace();
+        }
+
+        UserFileDaoImpl userDao = new UserFileDaoImpl();
+
+        User user = userDao.getUserByUsername(uname);
+        String destPage = "login.jsp";
+        String pass = user.getPassword();
+        String username = user.getUsername();
```

Comment:
I get a NullPointerException here when I submit with an invalid username.

I think you should put the variable assignments for pass and username in a try/catch block.
Also in this case for the catch block, maybe send an error message like you do in index.jsp